MODIS and SEADAS test failures

**Describe the bug**

MODIS and SEADAS tests are failing with latest versions of all libraries.

**To Reproduce**

```bash
mamba update --all
pytest reader_tests/modis_tests/ reader_tests/test_seadas_l2.py
```
**Expected behavior**

I expect all tests to pass.

**Actual results**

25 test failures:

```
================================================================== short test summary info ===================================================================
FAILED modis_tests/test_modis_l1b.py::TestModisL1b::test_load_longitude_latitude[modis_l1b_nasa_mod021km_file-True-False-False-1000] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l1b.py::TestModisL1b::test_load_longitude_latitude[modis_l1b_imapp_1000m_file-True-False-False-1000] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l1b.py::TestModisL1b::test_load_longitude_latitude[modis_l1b_nasa_mod02hkm_file-False-True-True-250] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l1b.py::TestModisL1b::test_load_longitude_latitude[modis_l1b_nasa_mod02qkm_file-False-True-True-250] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l1b.py::TestModisL1b::test_load_longitude_latitude[modis_l1b_nasa_1km_mod03_files-True-True-True-250] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l1b.py::TestModisL1b::test_load_sat_zenith_angle - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l1b.py::TestModisL1b::test_load_vis - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l1b.py::TestModisL1b::test_load_vis_saturation[False] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l1b.py::TestModisL1b::test_load_vis_saturation[True] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l2.py::TestModisL2::test_load_longitude_latitude[modis_l2_nasa_mod35_file-True-False-False-1000] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l2.py::TestModisL2::test_load_quality_assurance - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l2.py::TestModisL2::test_load_category_dataset[modis_l2_nasa_mod35_mod03_files-loadables0-1000-1000-True] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l2.py::TestModisL2::test_load_category_dataset[modis_l2_imapp_mask_byte1_geo_files-loadables1-None-1000-True] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l2.py::TestModisL2::test_load_250m_cloud_mask_dataset[modis_l2_nasa_mod35_file-False] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l2.py::TestModisL2::test_load_250m_cloud_mask_dataset[modis_l2_nasa_mod35_mod03_files-True] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l2.py::TestModisL2::test_load_l2_dataset[modis_l2_nasa_mod06_file-loadables0-5000-True-4.0] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l2.py::TestModisL2::test_load_l2_dataset[modis_l2_imapp_snowmask_file-loadables1-1000-False-1.0] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l2.py::TestModisL2::test_load_l2_dataset[modis_l2_imapp_snowmask_geo_files-loadables2-1000-True-1.0] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED modis_tests/test_modis_l3.py::TestModisL3::test_load_l3_dataset - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED test_seadas_l2.py::TestSEADAS::test_load_chlor_a[False-seadas_l2_modis_chlor_a-Aqua-exp_sensor0-10] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED test_seadas_l2.py::TestSEADAS::test_load_chlor_a[False-seadas_l2_viirs_npp_chlor_a-Suomi-NPP-exp_sensor1-16] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED test_seadas_l2.py::TestSEADAS::test_load_chlor_a[False-seadas_l2_viirs_j01_chlor_a-NOAA-20-exp_sensor2-16] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED test_seadas_l2.py::TestSEADAS::test_load_chlor_a[True-seadas_l2_modis_chlor_a-Aqua-exp_sensor0-10] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED test_seadas_l2.py::TestSEADAS::test_load_chlor_a[True-seadas_l2_viirs_npp_chlor_a-Suomi-NPP-exp_sensor1-16] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
FAILED test_seadas_l2.py::TestSEADAS::test_load_chlor_a[True-seadas_l2_viirs_j01_chlor_a-NOAA-20-exp_sensor2-16] - pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
========================================================= 25 failed, 16 passed, 1 warning in 14.61s ==========================================================
```

**Environment Info:**
 - OS: openSUSE Leap 15.3
 - Satpy Version: 0.51.0
 - Readers and writers dependencies (when relevant):

Extract:

```
modis_l1b:  ok
modis_l2:  ok
modis_l3:  ok
seadas_l2:  ok
```

**Additional context**

Must be some recent dependency that got upgraded, as it was working before I did `mamba update --all`.  The updates were:

```
     aiohttp  {3.9.5 (conda-forge/linux-64) -> 3.10.5 (conda-forge/linux-64)}                                                                                                                                                                                                                                        [67/216]
     attrs  {23.2.0 (conda-forge/noarch) -> 24.2.0 (conda-forge/noarch)}
     aws-c-auth  {0.7.22 (conda-forge/linux-64) -> 0.7.25 (conda-forge/linux-64)}
     aws-c-cal  {0.6.15 (conda-forge/linux-64) -> 0.7.3 (conda-forge/linux-64)}
     aws-c-common  {0.9.23 (conda-forge/linux-64) -> 0.9.27 (conda-forge/linux-64)}
     aws-c-compression  {0.2.18 (conda-forge/linux-64) -> 0.2.18 (conda-forge/linux-64)}
     aws-c-event-stream  {0.4.2 (conda-forge/linux-64) -> 0.4.3 (conda-forge/linux-64)}
     aws-c-http  {0.8.2 (conda-forge/linux-64) -> 0.8.7 (conda-forge/linux-64)}
     aws-c-io  {0.14.9 (conda-forge/linux-64) -> 0.14.18 (conda-forge/linux-64)}
     aws-c-mqtt  {0.10.4 (conda-forge/linux-64) -> 0.10.4 (conda-forge/linux-64)}
     aws-c-s3  {0.5.10 (conda-forge/linux-64) -> 0.6.4 (conda-forge/linux-64)}
     aws-c-sdkutils  {0.1.16 (conda-forge/linux-64) -> 0.1.19 (conda-forge/linux-64)}
     aws-checksums  {0.1.18 (conda-forge/linux-64) -> 0.1.18 (conda-forge/linux-64)}
     aws-crt-cpp  {0.26.12 (conda-forge/linux-64) -> 0.27.5 (conda-forge/linux-64)}
     aws-sdk-cpp  {1.11.329 (conda-forge/linux-64) -> 1.11.379 (conda-forge/linux-64)}
     azure-core-cpp  {1.12.0 (conda-forge/linux-64) -> 1.13.0 (conda-forge/linux-64)}
     azure-identity-cpp  {1.8.0 (conda-forge/linux-64) -> 1.8.0 (conda-forge/linux-64)}
     azure-storage-blobs-cpp  {12.11.0 (conda-forge/linux-64) -> 12.12.0 (conda-forge/linux-64)}
     azure-storage-common-cpp  {12.6.0 (conda-forge/linux-64) -> 12.7.0 (conda-forge/linux-64)}
     azure-storage-files-datalake-cpp  {12.10.0 (conda-forge/linux-64) -> 12.11.0 (conda-forge/linux-64)}
     backports  {1.0 (conda-forge/noarch) -> 1.0 (conda-forge/noarch)}
     bokeh  {3.4.2 (conda-forge/noarch) -> 3.5.1 (conda-forge/noarch)}
     boto3  {1.34.138 (conda-forge/noarch) -> 1.35.1 (conda-forge/noarch)}
     botocore  {1.34.138 (conda-forge/noarch) -> 1.35.2 (conda-forge/noarch)}
     bzip2  {1.0.8 (conda-forge/linux-64) -> 1.0.8 (conda-forge/linux-64)}
     c-ares  {1.28.1 (conda-forge/linux-64) -> 1.33.0 (conda-forge/linux-64)}
     cffi  {1.16.0 (conda-forge/linux-64) -> 1.17.0 (conda-forge/linux-64)}
     coverage  {7.5.4 (conda-forge/linux-64) -> 7.6.1 (conda-forge/linux-64)}
     curl  {8.8.0 (conda-forge/linux-64) -> 8.9.1 (conda-forge/linux-64)}
     dask  {2024.6.2 (conda-forge/noarch) -> 2024.8.1 (conda-forge/noarch)}
     dask-core  {2024.6.2 (conda-forge/noarch) -> 2024.8.1 (conda-forge/noarch)}
     dask-expr  {1.1.6 (conda-forge/noarch) -> 1.1.11 (conda-forge/noarch)}
     distributed  {2024.6.2 (conda-forge/noarch) -> 2024.8.1 (conda-forge/noarch)}
     exceptiongroup  {1.2.0 (conda-forge/noarch) -> 1.2.2 (conda-forge/noarch)}
     flake8  {7.1.0 (conda-forge/noarch) -> 7.1.1 (conda-forge/noarch)}
     fmt  {10.2.1 (conda-forge/linux-64) -> 11.0.2 (conda-forge/linux-64)}
     fonttools  {4.53.0 (conda-forge/linux-64) -> 4.53.1 (conda-forge/linux-64)}
     geos  {3.12.1 (conda-forge/linux-64) -> 3.12.2 (conda-forge/linux-64)}
     geotiff  {1.7.3 (conda-forge/linux-64) -> 1.7.3 (conda-forge/linux-64)}
     gettext  {0.22.5 (conda-forge/linux-64) -> 0.22.5 (conda-forge/linux-64)}
     gettext-tools  {0.22.5 (conda-forge/linux-64) -> 0.22.5 (conda-forge/linux-64)}
     git  {2.45.2 (conda-forge/linux-64) -> 2.46.0 (conda-forge/linux-64)}
     glib  {2.80.2 (conda-forge/linux-64) -> 2.80.3 (conda-forge/linux-64)}
     glib-tools  {2.80.2 (conda-forge/linux-64) -> 2.80.3 (conda-forge/linux-64)}
     graphviz  {11.0.0 (conda-forge/linux-64) -> 12.0.0 (conda-forge/linux-64)}
     gst-plugins-base  {1.24.5 (conda-forge/linux-64) -> 1.24.6 (conda-forge/linux-64)}
     gstreamer  {1.24.5 (conda-forge/linux-64) -> 1.24.6 (conda-forge/linux-64)}
     gtk2  {2.24.33 (conda-forge/linux-64) -> 2.24.33 (conda-forge/linux-64)}
     harfbuzz  {8.5.0 (conda-forge/linux-64) -> 9.0.0 (conda-forge/linux-64)}
     identify  {2.5.36 (conda-forge/noarch) -> 2.6.0 (conda-forge/noarch)}
     importlib-metadata  {8.0.0 (conda-forge/noarch) -> 8.4.0 (conda-forge/noarch)}
     importlib_metadata  {8.0.0 (conda-forge/noarch) -> 8.4.0 (conda-forge/noarch)}
     json-c  {0.17 (conda-forge/linux-64) -> 0.17 (conda-forge/linux-64)}
     libabseil  {20240116.2 (conda-forge/linux-64) -> 20240116.2 (conda-forge/linux-64)}
     libarrow  {16.1.0 (conda-forge/linux-64) -> 17.0.0 (conda-forge/linux-64)}
     libarrow-acero  {16.1.0 (conda-forge/linux-64) -> 17.0.0 (conda-forge/linux-64)}
     libarrow-dataset  {16.1.0 (conda-forge/linux-64) -> 17.0.0 (conda-forge/linux-64)}
     libarrow-flight  {16.1.0 (conda-forge/linux-64) -> 17.0.0 (conda-forge/linux-64)}
     libarrow-flight-sql  {16.1.0 (conda-forge/linux-64) -> 17.0.0 (conda-forge/linux-64)}
     libarrow-gandiva  {16.1.0 (conda-forge/linux-64) -> 17.0.0 (conda-forge/linux-64)}
     libarrow-substrait  {16.1.0 (conda-forge/linux-64) -> 17.0.0 (conda-forge/linux-64)}
     libasprintf  {0.22.5 (conda-forge/linux-64) -> 0.22.5 (conda-forge/linux-64)}
     libasprintf-devel  {0.22.5 (conda-forge/linux-64) -> 0.22.5 (conda-forge/linux-64)}
     libblas  {3.9.0 (conda-forge/linux-64) -> 3.9.0 (conda-forge/linux-64)}
     libboost-headers  {1.85.0 (conda-forge/linux-64) -> 1.86.0 (conda-forge/linux-64)}
     libcblas  {3.9.0 (conda-forge/linux-64) -> 3.9.0 (conda-forge/linux-64)}
     libclang  {18.1.8 (conda-forge/linux-64) -> 18.1.8 (conda-forge/linux-64)}
     libclang13  {18.1.8 (conda-forge/linux-64) -> 18.1.8 (conda-forge/linux-64)}
     libcurl  {8.8.0 (conda-forge/linux-64) -> 8.9.1 (conda-forge/linux-64)}
     libdeflate  {1.20 (conda-forge/linux-64) -> 1.21 (conda-forge/linux-64)}
     libgcrypt  {1.11.0 (conda-forge/linux-64) -> 1.11.0 (conda-forge/linux-64)}
     libgdal  {3.9.1 (conda-forge/linux-64) -> 3.9.2 (conda-forge/linux-64)}
     libgettextpo  {0.22.5 (conda-forge/linux-64) -> 0.22.5 (conda-forge/linux-64)}
     libgettextpo-devel  {0.22.5 (conda-forge/linux-64) -> 0.22.5 (conda-forge/linux-64)}
     libglib  {2.80.2 (conda-forge/linux-64) -> 2.80.3 (conda-forge/linux-64)}
     libgoogle-cloud  {2.25.0 (conda-forge/linux-64) -> 2.28.0 (conda-forge/linux-64)}
     libgoogle-cloud-storage  {2.25.0 (conda-forge/linux-64) -> 2.28.0 (conda-forge/linux-64)}
     libkml  {1.3.0 (conda-forge/linux-64) -> 1.3.0 (conda-forge/linux-64)}
     liblapack  {3.9.0 (conda-forge/linux-64) -> 3.9.0 (conda-forge/linux-64)}
     libllvm18  {18.1.8 (conda-forge/linux-64) -> 18.1.8 (conda-forge/linux-64)}
     libnl  {3.9.0 (conda-forge/linux-64) -> 3.10.0 (conda-forge/linux-64)}
     libopenblas  {0.3.27 (conda-forge/linux-64) -> 0.3.27 (conda-forge/linux-64)}
     libparquet  {16.1.0 (conda-forge/linux-64) -> 17.0.0 (conda-forge/linux-64)}
     libpq  {16.3 (conda-forge/linux-64) -> 16.4 (conda-forge/linux-64)}
     librsvg  {2.58.1 (conda-forge/linux-64) -> 2.58.3 (conda-forge/linux-64)}
     librttopo  {1.1.0 (conda-forge/linux-64) -> 1.1.0 (conda-forge/linux-64)}
     libspatialite  {5.1.0 (conda-forge/linux-64) -> 5.1.0 (conda-forge/linux-64)}
     libthrift  {0.19.0 (conda-forge/linux-64) -> 0.20.0 (conda-forge/linux-64)}
     libtiff  {4.6.0 (conda-forge/linux-64) -> 4.6.0 (conda-forge/linux-64)}
     libxml2  {2.12.7 (conda-forge/linux-64) -> 2.12.7 (conda-forge/linux-64)}
     matplotlib  {3.9.0 (conda-forge/linux-64) -> 3.9.1 (conda-forge/linux-64)}
     matplotlib-base  {3.9.0 (conda-forge/linux-64) -> 3.9.1 (conda-forge/linux-64)}
     mysql-common  {8.3.0 (conda-forge/linux-64) -> 8.3.0 (conda-forge/linux-64)}
     mysql-libs  {8.3.0 (conda-forge/linux-64) -> 8.3.0 (conda-forge/linux-64)}
     nss  {3.101 (conda-forge/linux-64) -> 3.103 (conda-forge/linux-64)}
     numcodecs  {0.12.1 (conda-forge/linux-64) -> 0.13.0 (conda-forge/linux-64)}
     numpy  {2.0.0 (conda-forge/linux-64) -> 2.0.1 (conda-forge/linux-64)}
     orc  {2.0.1 (conda-forge/linux-64) -> 2.0.2 (conda-forge/linux-64)}
     pango  {1.54.0 (conda-forge/linux-64) -> 1.54.0 (conda-forge/linux-64)}
     pcre2  {10.44 (conda-forge/linux-64) -> 10.44 (conda-forge/linux-64)}
     pip  {24.0 (conda-forge/noarch) -> 24.2 (conda-forge/noarch)}
     poppler  {24.04.0 (conda-forge/linux-64) -> 24.08.0 (conda-forge/linux-64)}
     postgresql  {16.3 (conda-forge/linux-64) -> 16.4 (conda-forge/linux-64)}
     pre-commit  {3.7.1 (conda-forge/noarch) -> 3.8.0 (conda-forge/noarch)}
     proj  {9.4.1 (conda-forge/linux-64) -> 9.4.1 (conda-forge/linux-64)}
     pure_eval  {0.2.2 (conda-forge/noarch) -> 0.2.3 (conda-forge/noarch)}
     pyarrow  {16.1.0 (conda-forge/linux-64) -> 17.0.0 (conda-forge/linux-64)}
     pyarrow-core  {16.1.0 (conda-forge/linux-64) -> 17.0.0 (conda-forge/linux-64)}
     pycodestyle  {2.12.0 (conda-forge/noarch) -> 2.12.1 (conda-forge/noarch)}
     pyproj  {3.6.1 (conda-forge/linux-64) -> 3.6.1 (conda-forge/linux-64)}
     pyresample  {1.28.4 (conda-forge/linux-64) -> 1.29.0 (conda-forge/linux-64)}
     pyspectral  {0.13.2 (conda-forge/noarch) -> 0.13.4 (conda-forge/noarch)}
     pytest  {8.2.2 (conda-forge/noarch) -> 8.3.2 (conda-forge/noarch)}
     python_abi  {3.12 (conda-forge/linux-64) -> 3.12 (conda-forge/linux-64)}
     pyyaml  {6.0.1 (conda-forge/linux-64) -> 6.0.2 (conda-forge/linux-64)}
     pyzmq  {26.0.3 (conda-forge/linux-64) -> 26.1.1 (conda-forge/linux-64)}
     qhull  {2020.2 (conda-forge/linux-64) -> 2020.2 (conda-forge/linux-64)}
     qt-main  {5.15.8 (conda-forge/linux-64) -> 5.15.8 (conda-forge/linux-64)}
     rasterio  {1.3.10 (conda-forge/linux-64) -> 1.3.10 (conda-forge/linux-64)}
     rdma-core  {52.0 (conda-forge/linux-64) -> 53.0 (conda-forge/linux-64)}
     rioxarray  {0.15.7 (conda-forge/noarch) -> 0.17.0 (conda-forge/noarch)}
     s2n  {1.4.16 (conda-forge/linux-64) -> 1.5.0 (conda-forge/linux-64)}
     scipy  {1.14.0 (conda-forge/linux-64) -> 1.14.0 (conda-forge/linux-64)}
     setuptools  {70.1.1 (conda-forge/noarch) -> 72.2.0 (conda-forge/noarch)}
     shapely  {2.0.4 (conda-forge/linux-64) -> 2.0.6 (conda-forge/linux-64)}
     snappy  {1.2.0 (conda-forge/linux-64) -> 1.2.1 (conda-forge/linux-64)}
     snuggs  {1.4.7 (conda-forge/noarch) -> 1.4.7 (conda-forge/noarch)}
     spdlog  {1.13.0 (conda-forge/linux-64) -> 1.14.1 (conda-forge/linux-64)}
     sphinx  {7.3.7 (conda-forge/noarch) -> 7.4.7 (conda-forge/noarch)}
     sphinxcontrib-applehelp  {1.0.8 (conda-forge/noarch) -> 2.0.0 (conda-forge/noarch)}
     sphinxcontrib-devhelp  {1.0.6 (conda-forge/noarch) -> 2.0.0 (conda-forge/noarch)}
     sphinxcontrib-htmlhelp  {2.0.5 (conda-forge/noarch) -> 2.1.0 (conda-forge/noarch)}
     sphinxcontrib-qthelp  {1.0.7 (conda-forge/noarch) -> 2.0.0 (conda-forge/noarch)}
     tiledb  {2.24.1 (conda-forge/linux-64) -> 2.25.0 (conda-forge/linux-64)}
     tqdm  {4.66.4 (conda-forge/noarch) -> 4.66.5 (conda-forge/noarch)}
     trollimage  {1.24.0 (conda-forge/linux-64) -> 1.25.0 (conda-forge/linux-64)}
     ucx  {1.16.0 (conda-forge/linux-64) -> 1.17.0 (conda-forge/linux-64)}
     wheel  {0.43.0 (conda-forge/noarch) -> 0.44.0 (conda-forge/noarch)}
     xarray  {2024.6.0 (conda-forge/noarch) -> 2024.7.0 (conda-forge/noarch)}
     zipp  {3.19.2 (conda-forge/noarch) -> 3.20.0 (conda-forge/noarch)}
     zstandard  {0.22.0 (conda-forge/linux-64) -> 0.23.0 (conda-forge/linux-64)}
    +aiohappyeyeballs-2.4.0 (conda-forge/noarch)
    +libgdal-core-3.9.2 (conda-forge/linux-64)
    +libgdal-fits-3.9.2 (conda-forge/linux-64)
    +libgdal-grib-3.9.2 (conda-forge/linux-64)
    +libgdal-hdf4-3.9.2 (conda-forge/linux-64)
    +libgdal-hdf5-3.9.2 (conda-forge/linux-64)
    +libgdal-jp2openjpeg-3.9.2 (conda-forge/linux-64)
    +libgdal-kea-3.9.2 (conda-forge/linux-64)
    +libgdal-netcdf-3.9.2 (conda-forge/linux-64)
    +libgdal-pdf-3.9.2 (conda-forge/linux-64)
    +libgdal-pg-3.9.2 (conda-forge/linux-64)
    +libgdal-postgisraster-3.9.2 (conda-forge/linux-64)
    +libgdal-tiledb-3.9.2 (conda-forge/linux-64)
    +libgdal-xls-3.9.2 (conda-forge/linux-64)
    +libllvm17-17.0.6 (conda-forge/linux-64)
    +xorg-libxxf86vm-1.1.5 (conda-forge/linux-64)
```

I'm confused by your first message @mraspaud. The traceback in the CI seems to suggest this is just us (satpy) requesting the `.info()` not dask. Right?

```
 satpy/readers/hdfeos_base.py:421: in get_dataset
    data = self._load_ds_by_name(dataset_name)
satpy/readers/hdfeos_base.py:385: in _load_ds_by_name
    return self.load_dataset(var_names)
satpy/readers/hdfeos_base.py:222: in load_dataset
    attrs=dataset.attributes())
../../../miniconda3/envs/test-environment/lib/python3.11/site-packages/pyhdf/SD.py:2786: in attributes
    natts = self.info()[4]
../../../miniconda3/envs/test-environment/lib/python3.11/site-packages/pyhdf/SD.py:2128: in info
    _checkErr('info', status, "cannot execute")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

procName = 'info', val = -1, msg = 'cannot execute'

    def _checkErr(procName, val, msg=""):
    
        if val is None or (not isinstance(val, str) and val < 0):
            #_C._HEprint();
            errCode = _C.HEvalue(1)
            if errCode != 0:
                err = "%s (%d): %s" % (procName, errCode, _C.HEstring(errCode))
            else:
                err = "%s : %s" % (procName, msg)
>           raise HDF4Error(err)
E           pyhdf.error.HDF4Error: info (59): Invalid arguments to routine
```

--- a/satpy/readers/hdf4_utils.py
+++ b/satpy/readers/hdf4_utils.py
@@ -18,10 +18,12 @@
 """Helpers for reading hdf4-based files."""
 
 import logging
+import os
 
 import dask.array as da
 import numpy as np
 import xarray as xr
+from dask.base import tokenize
 from pyhdf.SD import SD, SDC, SDS
 
 from satpy.readers.file_handlers import BaseFileHandler
@@ -45,12 +47,24 @@ HTYPE_TO_DTYPE = {
 }
 
 
-def from_sds(var, *args, **kwargs):
+def from_sds(var, src_path, *args, **kwargs):
     """Create a dask array from a SD dataset."""
-    var.__dict__["dtype"] = np.dtype(HTYPE_TO_DTYPE[var.info()[3]])
-    shape = var.info()[2]
+    var_info = var.info()
+    var.__dict__["dtype"] = np.dtype(HTYPE_TO_DTYPE[var_info[3]])
+    shape = var_info[2]
     var.__dict__["shape"] = shape if isinstance(shape, (tuple, list)) else tuple(shape)
-    return da.from_array(var, *args, **kwargs)
+
+    name = kwargs.pop("name", None)
+    if name is None:
+        var_name = var_info[0]
+        tokenize_args = (os.fspath(src_path), var_name)
+        if args:
+            tokenize_args += (args,)
+        if kwargs:
+            tokenize_args += (kwargs,)
+        # put variable name in the front for easier dask debugging
+        name = var_name + "-" + tokenize(*tokenize_args)
+    return da.from_array(var, *args, name=name, **kwargs)
 
 
 class HDF4FileHandler(BaseFileHandler):
@@ -92,7 +106,7 @@ class HDF4FileHandler(BaseFileHandler):
 
     def _open_xarray_dataset(self, val, chunks=CHUNK_SIZE):
         """Read the band in blocks."""
-        dask_arr = from_sds(val, chunks=chunks)
+        dask_arr = from_sds(val, self.filename, chunks=chunks)
         attrs = val.attributes()
         return xr.DataArray(dask_arr, dims=("y", "x"),
                             attrs=attrs)

--- a/satpy/readers/hdf4_utils.py
+++ b/satpy/readers/hdf4_utils.py
@@ -47,7 +47,7 @@ HTYPE_TO_DTYPE = {
 }
 
 
-def from_sds(var, src_path, *args, **kwargs):
+def from_sds(var, src_path, **kwargs):
     """Create a dask array from a SD dataset."""
     var_info = var.info()
     var.__dict__["dtype"] = np.dtype(HTYPE_TO_DTYPE[var_info[3]])
@@ -58,8 +58,6 @@ def from_sds(var, src_path, *args, **kwargs):
     if name is None:
         var_name = var_info[0]
         tokenize_args = (os.fspath(src_path), var_name)
-        if args:
-            tokenize_args += (args,)
         if kwargs:
             tokenize_args += (kwargs,)
         # put variable name in the front for easier dask debugging

--- a/satpy/readers/hdf4_utils.py
+++ b/satpy/readers/hdf4_utils.py
@@ -62,7 +62,7 @@ def from_sds(var, src_path, **kwargs):
             tokenize_args += (kwargs,)
         # put variable name in the front for easier dask debugging
         name = var_name + "-" + tokenize(*tokenize_args)
-    return da.from_array(var, *args, name=name, **kwargs)
+    return da.from_array(var, name=name, **kwargs)
 
 
 class HDF4FileHandler(BaseFileHandler):

--- a/satpy/readers/hdfeos_base.py
+++ b/satpy/readers/hdfeos_base.py
@@ -216,7 +216,7 @@ class HDFEOSBaseFileReader(BaseFileHandler):
 
         dataset = self._read_dataset_in_file(dataset_name)
         chunks = self._chunks_for_variable(dataset)
-        dask_arr = from_sds(dataset, chunks=chunks)
+        dask_arr = from_sds(dataset, self.filename, chunks=chunks)
         dims = ("y", "x") if dask_arr.ndim == 2 else None
         data = xr.DataArray(dask_arr, dims=dims,
                             attrs=dataset.attributes())

--- a/satpy/readers/modis_l1b.py
+++ b/satpy/readers/modis_l1b.py
@@ -117,7 +117,7 @@ class HDFEOSBandReader(HDFEOSBaseFileReader):
         var_attrs = subdata.attributes()
         uncertainty = self.sd.select(var_name + "_Uncert_Indexes")
         chunks = self._chunks_for_variable(subdata)
-        array = xr.DataArray(from_sds(subdata, chunks=chunks)[band_index, :, :],
+        array = xr.DataArray(from_sds(subdata, self.filename, chunks=chunks)[band_index, :, :],
                              dims=["y", "x"]).astype(np.float32)
         valid_range = var_attrs["valid_range"]
         valid_min = np.float32(valid_range[0])
@@ -214,7 +214,7 @@ class HDFEOSBandReader(HDFEOSBaseFileReader):
         if not self._mask_saturated:
             return array
         uncertainty_chunks = self._chunks_for_variable(uncertainty)
-        band_uncertainty = from_sds(uncertainty, chunks=uncertainty_chunks)[band_index, :, :]
+        band_uncertainty = from_sds(uncertainty, self.filename, chunks=uncertainty_chunks)[band_index, :, :]
         array = array.where(band_uncertainty < 15)
         return array
 

--- a/satpy/readers/modis_l2.py
+++ b/satpy/readers/modis_l2.py
@@ -111,7 +111,7 @@ class ModisL2HDFFileHandler(HDFEOSGeoReader):
     def _select_hdf_dataset(self, hdf_dataset_name, byte_dimension):
         """Load a dataset from HDF-EOS level 2 file."""
         dataset = self.sd.select(hdf_dataset_name)
-        dask_arr = from_sds(dataset, chunks=CHUNK_SIZE)
+        dask_arr = from_sds(dataset, self.filename, chunks=CHUNK_SIZE)
         attrs = dataset.attributes()
         dims = ["y", "x"]
         if byte_dimension == 0:

